control_panel_api/settings/test.py

```diff
@@ -0,0 +1,9 @@
+from .base import *
```

Comment:
Interesting, so maybe we may have an if in the main `settings.py` to check if the `ENV` is `test` and mock automatically in that case.

So if someone forget to override the settings it would pick that we're in test environment and swap the boto3 client transparently.

Reply:
Also, in theory it's possible to stub `boto3.client` calls to return the mock right?

What I'm thinking is, maybe we can do this so that when the tests are run `boto3.client()` would return a mock.

Reply:
Another thing I'm realising is that we'd ideally want a way to set `ENV="test"` when running `python manage.py test` (don't know if there is a way to plug into it).

Reply:
I hear what you're saying about the manage test thing.  The usual practice is like this though where a developer can see the settings files are separated and named as test.py this then needs to be specified.  Between us here it's not an issue as I've just notified you all, and a new dev can read a line in the README about running them.

Another thing is we haven't really configured environments yet e.g. prod or local. We could default local to using mock so to explcitly hit aws you have to specify it via `--settings`, we could even create a settings file like `local_aws.py` that connects to it.

So this PR fixes our test issues, but now we have to think about local dev.  Explicitly have to use settings with actual aws?  Also what about a staging/dev aws environment to connect to? @andyhd

Reply:
Perhaps just this: https://stackoverflow.com/a/7651002

Reply:
The second answer to that stackoverflow question shows how easy it would be to override the TestRunner too.

Reply:
@andyhd good one. I think that's what I had in mind as it greatly reduce the chances of error (which is the point of these changes in the first place).

I think just add a note in the `README` wouldn't cut. A new dev comes in, see it's a django app then would run the test with the standard `python manage.py test`, let's say he already had AWS credentials in place it may _potentially_ hit the real service.